File: 0078-subsets/0078-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        
        
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        // vector<int> flag(nums.size(),0);
        fun(nums,0,ans,temp);
        return ans;
    }
        
     public static void fun(int[] nums,int ind,List<List<Integer>> ans,List<Integer> temp){
        if(ind==nums.length){
            ans.add(new ArrayList<>(temp));
            return;
        }
        temp.add(nums[ind]);
        fun(nums,ind+1,ans,temp);

        temp.remove(temp.size()-1);
        fun(nums,ind+1,ans,temp);

    }
    
}

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    void fun(vector<int> &nums,int ind,vector<vector<int>> &ans,vector<int> &temp){
        if(ind==nums.size()){
            ans.push_back(temp);
            return;
        }
        temp.push_back(nums[ind]);
        fun(nums,ind+1,ans,temp);
        
        temp.pop_back();
        fun(nums,ind+1,ans,temp);
        
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> temp;
        // vector<int> flag(nums.size(),0);
        fun(nums,0,ans,temp);
        return ans;
    }
};